Userland/Libraries/LibGUI/DynamicWidgetContainer.cpp

```diff
@@ -0,0 +1,475 @@
+    }
+
+    auto controls_color = m_hover_over_control_indicator ? palette().threed_shadow2() : palette().hover_highlight().lightened(0.7f);
+    auto expand_collapse_rect = Gfx::IntRect({}, s_expand_bitmap.size()).centered_within(m_expand_collapse_button);
+    if (view_state() == ViewState::Collapsed)
+        painter.draw_bitmap(expand_collapse_rect.location(), s_expand_bitmap, controls_color);
+
+    if (view_state() == ViewState::Expanded)
+        painter.draw_bitmap(expand_collapse_rect.location(), s_collapsed_bitmap, controls_color);
+
+    auto detach_rect = Gfx::IntRect({}, s_detach_bitmap.size()).centered_within(m_detach_widget_button);
+    painter.draw_bitmap(detach_rect.location(), s_detach_bitmap, controls_color);
+
+    auto paint_knurl = [&](int x, int y) {
+        painter.set_pixel(x, y, palette().threed_shadow1());
+        painter.set_pixel(x + 1, y, palette().threed_shadow1());
+        painter.set_pixel(x, y + 1, palette().threed_shadow1());
+        painter.set_pixel(x + 1, y + 1, palette().threed_highlight());
+    };
+    for (int i = 0; i < 5; ++i)
+        paint_knurl(1, 0 + (i * 3));
+}
+
+void DynamicWidgetContainer::second_paint_event(PaintEvent&)
+{
+    if (!m_is_dragging && !m_render_as_move_target)
+        return;
+
+    GUI::Painter painter(*this);
+    if (m_is_dragging) {
+        // FIXME: Would be nice if we could paint outside our own boundaries.
+        auto move_widget_indicator = rect().translated(m_current_mouse_position).translated(-m_drag_start_location);
+        painter.fill_rect(move_widget_indicator, palette().accent().with_alpha(50));
+        painter.draw_rect_with_thickness(move_widget_indicator, palette().accent(), 2);
+    } else if (m_render_as_move_target)
+        painter.fill_rect(rect(), palette().accent().with_alpha(50));
+}
+
+ErrorOr<void> DynamicWidgetContainer::detach_widgets()
+{
+    if (m_detached_widgets_window.is_null()) {
+        m_detached_widgets_window = TRY(GUI::Window::try_create());
+        m_detached_widgets_window->set_title(section_label().to_deprecated_string());
+        m_detached_widgets_window->set_window_type(WindowType::Normal);
+        if (m_detached_size.has_value())
+            m_detached_widgets_window->resize(detached_size());
+        else
+            m_detached_widgets_window->resize(640, 480);
+
+        m_detached_widgets_window->center_on_screen();
+
+        auto root_container = m_detached_widgets_window->set_main_widget<GUI::Frame>();
+        root_container->set_fill_with_background_color(true);
+        root_container->set_layout<GUI::VerticalBoxLayout>();
+        root_container->set_frame_style(Gfx::FrameStyle::Window);
+        auto transfer_childs = [this](auto reciever, auto childs) {
+            for (RefPtr<GUI::Widget> widget : childs) {
+                if (widget.ptr() == this->m_controls_spacer)
+                    continue;
+                widget->remove_from_parent();
+                widget->set_visible(true);
```

Comment:
Was thinking about it, where is already solved that on pixel paint.... that kind of widgets would need to be wrapped in another widget, so that the visibility does not get screwed up by the container.